### DISCUSSION TITLE
When using @Check decorator the migration is overwritten

### DISCUSSION THREAD

**Discussion by DiegoCalleri** [mikro-orm/mikro-orm#6314] [General]
### Describe the bug

Describe the bug
I added  constrain via @Check_decorator to our db like this, it will always drop and recreate the constrain again

```ts
@Entity({ tableName: 'poly', repository: () => PolyRepo})
@Check({ expression: `"poly_number" ~* '^[a-zA-Z0-9]+( [a-zA-Z0-9]+)?$' AND LENGTH("poly_number") <= 15`, name: 'poly_check' })
export class PolyEntity extends PolyEntity {
    @Property({ columnType: 'int', persist: false })
    @IsId()
    PolyId!: number;

    @Property({ columnType: 'varchar(15)', comment: 'Poly number' })
    @Length(15)
    @Matches(/^[a-zA-Z0-9]+( [a-zA-Z0-9]+)?$/)
    polyNumber!: string;
   ....
```
if i call migration

```shell
npx mikro-orm migration:create
```

result will always be

```ts
this.addSql('alter table "poly" drop constraint poly_check;');
this.addSql('alter table "poly" add constraint poly_check("poly_number" ~* \'^[a-zA-Z0-9]+( [a-zA-Z0-9]+)?$\' AND LENGTH("poly_number") <= 15);');
```

Versions

| Dependency | Version |
"@mikro-orm/core": "^6.1.12",
"@mikro-orm/migrations": "^6.1.12",
"@mikro-orm/nestjs": "^5.2.3",
"@mikro-orm/postgresql": "^6.1.12",
 "@mikro-orm/cli": "^6.1.12",
"typescript": "^5.1.3"

### Reproduction

This is an internal project

### What driver are you using?

@mikro-orm/postgresql

### MikroORM version

6.1.12

### Node.js version

18.17.1

### Operating system

Windows

### Validations

- [x] Read the [Contributing Guidelines](https://github.com/mikro-orm/mikro-orm/blob/master/CONTRIBUTING.md).
- [X] Read the [docs](https://mikro-orm.io/docs).
- [X] Check that there isn't [already an issue](https://github.com/mikro-orm/mikro-orm/issues) that reports the same bug to avoid creating a duplicate.
- [X] Check that this is a concrete bug. For Q&A open a [GitHub Discussion](https://github.com/mikro-orm/mikro-orm/discussions) or join our [Discord](https://discord.gg/w8bjxFHS7X).
- [X] The provided reproduction is a [minimal reproducible example](https://stackoverflow.com/help/minimal-reproducible-example) of the bug.

**Comment by B4nan** [comment]
Such problems are usually in malformed expression, it needs to match 100% to the normalized form in the database. You can use verbose logging to see what the exact shape is:

https://mikro-orm.io/docs/migrations#debugging

Also, your first step before reporting anything should be upgrading to latest version, to ensure it's not something already fixed, 6.1.12 is quite old.

---

(Could be also something with the snapshot file, I will have a closer look to ensure what its about.)

**Reply by B4nan** [reply]
I am not able to repreduce this with the latest version, if you can, complete repro welcome.
